src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-polly/index.mdx

```diff
@@ -0,0 +1,233 @@
+
+Define the function handler by creating a new file, `amplify/data/convertTextToSpeech.ts`. This function converts text into speech using Amazon Polly and stores the synthesized speech as an MP3 file in an S3 bucket.
+
+```ts title="amplify/data/convertTextToSpeech.ts"
+
+import { Schema } from "./resource";
+import { PollyClient, StartSpeechSynthesisTaskCommand } from '@aws-sdk/client-polly'
+import { env } from '$amplify/env/convertTextToSpeech'
+
+export const handler: Schema["convertTextToSpeech"]["functionHandler"] = async (event) => {
+  const client = new PollyClient()
+  const task = new StartSpeechSynthesisTaskCommand({
+    OutputFormat: 'mp3',
+    SampleRate: '8000',
+    Text: event.arguments.text,
+    TextType: 'text',
+    VoiceId: 'Amy',
+    OutputS3BucketName: env.predictions_gen2_BUCKET_NAME,
+    OutputS3KeyPrefix: 'public/'
+  })
+  const result = await client.send(task)
+
+  return result.SynthesisTask?.OutputUri?.replace(
+    'https://s3.us-east-1.amazonaws.com/' + env.predictions_gen2_BUCKET_NAME + '/public/', "") ?? ""
+}
+
+```
+
+## Step 6 - Define the Custom Mutation and function
+
+In your `amplify/data/resource.ts` file, define the function using defineFunction and then reference the function with your mutation using a.handler.function() as a handler.
+
+```ts title="amplify/data/resource.ts"
+import { type ClientSchema, a, defineData, defineFunction } from '@aws-amplify/backend';
+
+export const convertTextToSpeech = defineFunction({
+  entry: './convertTextToSpeech.ts'
+})
+
+const schema = a.schema({
+  convertTextToSpeech: a.mutation()
+    .arguments({
+      text: a.string().required()
+    })
+    .returns(a.string().required())
+    .authorization([a.allow.public()])
+    .handler(a.handler.function(convertTextToSpeech))
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: 'apiKey',
+    // API Key is used for a.allow.public() rules
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30,
+    },
+  },
+});
+
+
+```
+## Step 7 - Update Storage permissions
+
+Customize your storage settings to manage access to various paths within your storage bucket. It's necessary to update the Storage resource to provide access to the `convertTextToSpeech` resource. Modify the file `amplify/storage/resource.ts` as shown below.
+
+```ts title="amplify/storage/resource.ts"
+import { defineStorage } from "@aws-amplify/backend"
+import { convertTextToSpeech } from "../data/resource"
+
+export const storage = defineStorage({
+  name: "predictions_gen2",
+  access: allow => ({
+    '/public/*': [
+      allow.resource(convertTextToSpeech).to(['write']),
+      allow.guest.to(['read', 'write', 'get'])
+    ]
+  })
+})
+```
+
+## Step 8 - Configure the frontend
+
+Import and load the configuration file in your app. It's recommended you add the Amplify configuration step to your app's root entry point. 
```

Comment:
this text seems to be copied/pasted into the wrong place. The code below doesn't show the configuration file.

Reply:
added the config code